src/moscot/plotting/_utils.py

```diff
@@ -359,45 +360,137 @@ def _plot_temporal(
     show: bool = False,
+    suptitle_fontsize: Optional[float] = None,
     **kwargs: Any,
-) -> mp.figure.Figure:
-    all_keys = adata.obs[temporal_key].unique()
-    if time_points is None:
-        constant_fill_keys: Set[K] = set()
+) -> mpl.figure.Figure:
+
+    fig, axs = plt.subplots(
+        1, 1 if time_points is None else len(time_points), figsize=figsize, dpi=dpi, constrained_layout=True
+    )
+    axs = np.ravel(axs)  # make into iterable
+
+    if not push and time_points is not None:
+        time_points = time_points[::-1]
+
+    if isinstance(title, list):
+        if TYPE_CHECKING:
+            assert isinstance(time_points, list)
+        if len(title) != len(time_points):
+            raise ValueError("If `title` is a list, its length must be equal to the length of `time_points`.")
+        titles = title
     else:
-        constant_fill_keys = set(all_keys) - set(time_points)
-    tmp = np.full(len(adata), np.nan)
-    for t in adata.obs[temporal_key].unique():
-        mask = adata.obs[temporal_key] == t
-        if t in constant_fill_keys:
-            tmp[mask] = constant_fill_value
+        name = "descendants" if push else "ancestors"
+        if time_points is not None:
+            titles = [f"{categories} at time point {start if push else end}"]
+            titles.extend([f"{name} at time point {time_points[i]}" for i in range(1, len(time_points))])
         else:
-            tmp[mask] = adata[adata.obs[temporal_key] == t].obs[key_stored]
-
-    adata.obs[result_key] = tmp
-
-    sc.set_figure_params(figsize=figsize, dpi=dpi)  # TODO(@MUCDK, michalk8): necessary? want to make it uniform
-    fig = sc.pl.embedding(
-        adata=adata,
-        basis=basis,
-        color=result_key,
-        color_map=cont_cmap,
-        title=title,
-        ax=ax,
-        show=show,
-        **kwargs,
-    )
+            titles = [f"{categories} at time point {start if push else end} and {name}"]
+    for i, ax in enumerate(axs):
+        if time_points is None:
+            if scale:
+                vmin, vmax = np.nanmin(adata.obs[key_stored]), np.nanmax(adata.obs[key_stored])
+            else:
+                vmin, vmax = 0, 1
+            adata.obs[f"result_key_{i}"] = (adata.obs[key_stored] - vmin) / (vmax - vmin)
+            size = None
+        else:
+            tmp = np.full(len(adata), constant_fill_value)
+            mask = adata.obs[temporal_key] == time_points[i]
+
+            tmp[mask] = adata[adata.obs[temporal_key] == time_points[i]].obs[key_stored]
+            if scale:
+                vmin = np.nanmin(adata.obs[key_stored] * mask)
+                vmax = np.nanmax(adata.obs[key_stored] * mask)
+            else:
+                vmin = 0
+                vmax = 1
+            adata.obs[f"result_key_{i}"] = (tmp - vmin) / (vmax - vmin)
+            size = (mask * 120000 * dot_scale_factor + (1 - mask) * 120000) / adata.n_obs  # 120,000 from scanpy
+
+            _ = kwargs.pop("color_map", None)
```

Comment:
Would pop this not in the loop.

Reply:
We set it in the loop below, which is why we have to remove it again in the next iteration of the loop.